File: nodejs-microservices-tutorial/src/infrastructure/messaging/setupConsumers.ts
import { RabbitMQConsumer } from './RabbitMQConsumer';
import { UserEventHandler } from '../../application/services/UserEventHandler';

export async function setupConsumers(): Promise<void> {
  const consumer = new RabbitMQConsumer();
  await consumer.connect();

  const eventHandler = new UserEventHandler();

  // Consume user.created events
  await consumer.consume(
    process.env.RABBITMQ_QUEUE || 'user_notifications',
    async (message) => {
      if (message.routingKey === 'user.created') {
        await eventHandler.handleUserCreated(message);
      }
    }
  );

  console.log('Consumers setup complete');
}

File: fastapi-microservices-tutorial/app/infrastructure/messaging/setup_consumers.py
import os
from app.infrastructure.messaging.rabbitmq_consumer import RabbitMQConsumer
from app.application.services.user_event_handler import UserEventHandler

async def setup_consumers():
    """Setup message consumers"""
    consumer = RabbitMQConsumer()
    await consumer.connect()

    event_handler = UserEventHandler()

    # Consume user.created events
    async def handle_message(message: dict):
        routing_key = message.get("routing_key", "")
        if routing_key == "user.created":
            await event_handler.handle_user_created(message)

    await consumer.consume(
        os.getenv("RABBITMQ_QUEUE", "user_notifications"),
        handle_message
    )

    print("Consumers setup complete")
